pennylane/tape/reversible.py

```diff
@@ -1,273 +1,273 @@
-            qml.QubitStateVector(final_state, wires=dev_wires)
-
-            # evolve circuit backwards until gate we want to differentiate
-            for op in between_ops_inverse:
-                op.queue().inv()
-
-            # apply generator needed for differentiation
-            qml.apply(generator)
-
-            # evolve forwards again
-            for op in between_ops:
-                op.queue()
-
-            qml.state()
-
-        tapes = [new_circuit]
-
-        def processing_fn(results):
-            """Computes the gradient of the parameter at index idx via the
-            reversible backprop method.
-
-            Args:
-                results (list[real]): evaluated quantum tapes
-
-            Returns:
-                array[float]: 1-dimensional array of length determined by the tape output
-                measurement statistics
-            """
-            dstate = results[0][0]
-
-            # compute matrix element <d(state)|O|state> for each observable O
-            # TODO: if all observables act on same number of wires, could do all at once with einsum
-            matrix_elems = [
-                self._matrix_elem(dstate, ob, final_state, dev_wires) for ob in self.observables
-            ]
-            matrix_elems = np.array(matrix_elems)
-            return 2 * multiplier * np.imag(matrix_elems)
-
-        return tapes, processing_fn
-
-    def jacobian(self, device, params=None, **options):
-        # The reversible_diff method needs to evaluate the circuit
-        # at the unshifted parameter values; the pre-rotated statevector is then stored
-        # in the self._state attribute. Here, we set the value of the attribute to None
-
-        # before each Jacobian call, so that the statevector is calculated only once.
-        self._final_state = None
-        if device.shots is not None:
-            warnings.warn(
-                "Requested reversible differentiation to be computed with finite shots."
-                " Reversible differentiation always calculated exactly.",
-                UserWarning,
-            )
-
-        return super().jacobian(device, params, **options)
-
-    def analytic_pd(self, idx, params, **options):
-        device = options["device"]
-
-        # circuits constructed in reversible differentiation always start
-        # with the final state of the original circuit, which we store here
-        if self._final_state is None:
-            self.execute_device(params, device)
-            # todo: better create a new tape that has state as output method here?
-            self._final_state = device._pre_rotated_state.flatten()
-
-        # we need the wires to prepare the final state in each run, and to
-        # be able to compute an expecation value by hand
-        options["dev_wires"] = device.wires
-
-        return self.reversible_diff(idx, params=params, **options)
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
```

Comment:
@albi3ro looks like something went wrong in the diff for this file?

Reply:
Something similar happened to me before as well, I had to check out the file from op-refactor branch and redo any necessary changes...